### PR TITLE
Basic Sendgrid Transport Class

### PR DESCRIPTION
**Pull request by tsmgeek**
Basic SendgridSmtp class to support setting custom options via headers and also basic flag for click tracking.
We used this in production before we moved to a SendgridApi class but that has a dependency on the Sendgrid PHP Library which I am not sure if you would allow it, maybe by using a suggested dependency in composer?

### CODE REVIEW

File: library/Zend/Mail/Transport/SendgridSmtp.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
+<?php
+
+class WN_Mail_Transport_SendgridSmtp extends WN_Mail_Transport_Smtp
```
Comment (glensc) [inline library/Zend/Mail/Transport/SendgridSmtp.php:3]: wrong namespace

File: library/Zend/Mail/Transport/SendgridSmtp.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+     * API Options
+     *
+     * @var array
+     */
+    private $apiOptions = [];
+
+    public function __construct($options)
+    {
+         if (!isset($options['host'])) {
```
Comment (glensc) [inline library/Zend/Mail/Transport/SendgridSmtp.php:21]: inconsistent indent

File: library/Zend/Mail/Transport/SendgridSmtp.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+             throw new Zend_Application_Resource_Exception(
+                 'A host is necessary for smtp transport,'
+                 . ' but none was given'
+             );
+         }
+         $this->setClickTracking($options['clicktrack'] ?? false);
```
Comment (glensc) [inline library/Zend/Mail/Transport/SendgridSmtp.php:27]: missing implementation

File: library/Zend/Mail/Transport/SendgridSmtp.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        $options['ssl'] = true;
+        parent::__construct($options);
+    }
+
+    /**
+     * Set ClickTracking status
+     *
+     * @param bool $status
+     * @return void
+     *
```
Comment (glensc) [inline library/Zend/Mail/Transport/SendgridSmtp.php:40]: why is this commented out?

File: library/Zend/Mail/Transport/SendgridSmtp.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
+<?php
+
+class WN_Mail_Transport_SendgridSmtp extends WN_Mail_Transport_Smtp
+{
+    /**
+     * ClickTracking status
+     *
+     * @var bool
+     */
+    protected $clickTracking = false;
```
Comment (glensc) [inline library/Zend/Mail/Transport/SendgridSmtp.php:10]: shouldn't this be private?

File: library/Zend/Mail/Transport/SendgridSmtp.php [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        return $this->apiOptions;
+    }
+
+    /**
+     * Set API options
+     *
+     * @param   array $options
+     * @return  void
+     * 
+    public function setOptionsArray(array $options)
```
Comment (glensc) [inline library/Zend/Mail/Transport/SendgridSmtp.php:96]: rename to setOptions?

File: library/Zend/Mail/Transport/SendgridSmtp.php (hunk 5 above)
Comment (tsmgeek) [reply library/Zend/Mail/Transport/SendgridSmtp.php:10]: my thought was just if someone wanted to extend it, they could do that via the constructor though

**Pull request merged**